src/sign.rs

```diff
@@ -1,61 +1,244 @@
 use crate::account;
-use anyhow::Result;
+use anyhow::{bail, Context, Result};
+use clap::{Args, Subcommand};
 use fuel_crypto::{Message, SecretKey, Signature};
 use fuel_types::Bytes32;
 use rpassword::prompt_password;
-use std::{path::Path, str::FromStr};
+use std::{
+    path::{Path, PathBuf},
+    str::FromStr,
+};
 
-fn sign_transaction(
-    tx_id: Bytes32,
-    account_index: usize,
-    password: &str,
-    path: &Path,
-) -> Result<Signature> {
-    let secret_key = account::derive_secret_key(path, account_index, password)?;
-    sign_transaction_with_private_key(tx_id, secret_key)
+/// Sign some data (e.g. a transaction ID, a file, a string, or a hex-string)
+/// using either a wallet account or a private key.
+#[derive(Debug, Args)]
+pub struct Sign {
+    /// Sign using the wallet account at the given index.
+    /// Uses a discrete interactive prompt for password input.
+    #[clap(long, value_name = "ACCOUNT_INDEX")]
+    pub account: Option<usize>,
+    /// Sign using a private key.
+    /// Uses a discrete interactive prompt for collecting the private key.
+    #[clap(long)]
+    pub private: bool,
+    /// Sign by passing the private key directly.
+    ///
+    /// WARNING: This is primarily provided for non-interactive testing. Using this flag is
+    /// prone to leaving your private key exposed in your shell command history!
```

Comment:
Dumb question, but is it not possible to provide this option(signing directly with a private key) with a more secure method such as that used for the account signing option? 
If not I would almost prefer to not expose this method, unless there's an explicit warning to this effect displayed in the CLI when trying to use this.

Reply:
@nfurfaro 
- I agree this is risky (i.e., leaving secret keys in shell history)
- Ideally we'd implement this functionality (use private key directly) and protect oblivious users from themselves
- But this implementation includes a nice/clear warning, and it's opt-in - so I think this is fine (we can always circle back later)
  - @mitchmindtree Not a blocker, but maybe we could even scream at the user in their shell when they use this flag just to be extra sure they see it lol

Reply:
@nfurfaro the `--private` flag will allow for entering the private key in a password-style prompt interactively. As the comment mentions, this `--private-key` flag where we pass the key as an argument is more aimed at supporting non-interactive tests.

I was a little unsure about the `--private` flag name - perhaps it should be `--private-key` and the this flag should have a `--private-key-by-arg` or `--private-key-unsafe` or something?